src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -1196,12 +1217,16 @@ public SchemaShape GetOutputSchema(SchemaShape inputSchema)
                 if (col.ItemType != expectedType)
                     throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", input, expectedType.ToString(), col.ItemType.ToString());
             }
-            for (var i = 0; i < _options.OutputColumns.Length; i++)
-            {
-                resultDic[_options.OutputColumns[i]] = new SchemaShape.Column(_options.OutputColumns[i],
-                    _outputTypes[i].IsKnownSizeVector() ? SchemaShape.Column.VectorKind.Vector
-                    : SchemaShape.Column.VectorKind.VariableVector, _outputTypes[i].GetItemType(), false);
-            }
+
+            resultDic[_options.OutputColumns[0]] = new SchemaShape.Column(_options.OutputColumns[0],
+                    SchemaShape.Column.VectorKind.Vector, _outputTypes[0].GetItemType(), false);
+
+            var metadata = new List<SchemaShape.Column>();
+            metadata.Add(new SchemaShape.Column(AnnotationUtils.Kinds.KeyValues, SchemaShape.Column.VectorKind.Vector, TextDataViewType.Instance, false, inputSchema[1].Annotations));
```

Comment:
>, inputSchema[1].Annotations [](start = 151, length = 28)

This is not needed.

Reply:
By "this", I mean passing `inputSchema[1].Annotations` to the `SchemaShape.Column` constructor. The annotations of a `SchemaShape.Column` are stored themselves as a `SchemaShape` (since they are similar, in that they have a name and a type). The column you are instantiating here is a column in the `SchemaShape` of the annotations of the `PredictedLabel` column, it does not have annotations of its own.

---
In reply to: [327485460](https://github.com/dotnet/machinelearning/pull/4228#discussion_r327485460) [](ancestors = 327485460)

Reply:
Reactivating, since line 1284 still instantiates the new `SchemaShape.Column` incorrectly in iteration 16.

---
In reply to: [329082925](https://github.com/dotnet/machinelearning/pull/4228#discussion_r329082925) [](ancestors = 329082925,327485460)